src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -122,6 +124,50 @@ function matrix(A::Array{AbstractAlgebra.Generic.FreeModuleElem{T},1}) where T <
    return X
 end
 
+"""
+    upper_triangular_matrix(L::AbstractVector{T}) where T <: RingElem
+
+Return the upper triangular matrix whose non-zero entries (ordered lexicographically) are the elements of the sequence `L`.
+
+An error is returned whenever the length of `L` has not the form `n*(n+1)/2` for some integer `n`.
+"""
+function upper_triangular_matrix(L::AbstractVector{T}) where T <: RingElem
```

Comment:
The description refers to a sequence, and indeed, you could easily generalize this to work with arbitrary sequences of RingElem, including e.g. `fmpz_mat` and other Oscar vector-as-matrix types...
```suggestion
function upper_triangular_matrix(L)
   T = eltype(L)
   @assert T <: RingElem "L must be a collection of ring elements"
```